src/core/mac/mac.cpp

```diff
@@ -2264,5 +2290,49 @@ void Mac::LogFrameTxFailure(const Frame &, otError) const
 
 #endif // #if (OPENTHREAD_CONFIG_LOG_LEVEL >= OT_LOG_LEVEL_INFO) && (OPENTHREAD_CONFIG_LOG_MAC == 1)
 
+#if OPENTHREAD_CONFIG_ENABLE_TIME_SYNC
+void Mac::ProcessTimeIe(Frame &aFrame)
+{
+    TimeIe *timeIe = reinterpret_cast<TimeIe *>(aFrame.GetTimeIe());
+
+    VerifyOrExit(timeIe != NULL);
+
+    aFrame.SetNetworkTimeOffset(static_cast<int64_t>(timeIe->GetTime()) - static_cast<int64_t>(aFrame.GetTimestamp()));
+    aFrame.SetTimeSyncSeq(timeIe->GetSequence());
+
+exit:
+    return;
+}
+
+uint8_t Mac::GetTimeIeOffset(Frame &aFrame)
+{
+    uint8_t  offset = 0;
+    uint8_t *base   = aFrame.GetPsdu();
+    uint8_t *cur    = NULL;
+
+    cur = aFrame.GetTimeIe();
+    VerifyOrExit(cur != NULL);
+
+    cur += sizeof(VendorIeHeader);
+    offset = cur - base;
+
+exit:
+    return offset;
+}
+
+extern "C" void otPlatRadioFrameUpdated(otInstance *aInstance, otRadioFrame *aFrame)
+{
+    Instance *instance = static_cast<Instance *>(aInstance);
+    Frame     frame    = *static_cast<Frame *>(aFrame);
+
+    VerifyOrExit(instance->IsInitialized());
+
+    instance->GetThreadNetif().GetMac().ProcessTransmitSecurity(frame);
```

Comment:
Upon reviewing this again, I'm a bit concerned about `Mac::ProcessTransmitSecurity()` being called in interrupt context.  In particular, `Mac::ProcessTransmitSecurity()` seems to call out to other things (e.g. `KeyManager::IncrementMacFrameCounter()`.

I would prefer having `otPlatRadioFrameUpdated()` only read/write data passed in via `aFrame`.  More specifically:
1. Prepare the frame, including MAC Auxiliary Header before calling `otPlatRadioTransmit()`.
2. Radio driver updates the frame by writing the time value into the IE and calls `otPlatRadioFrameUpdated()`.
3. `otPlatRadioFrameUpdated()` performs AES-CCM by only reading/writing information in `aFrame`.

In general, OpenThread APIs should not be called in interrupt context.  `otPlatRadioFrameUpdated()` is an exception here and we should make that really clear.  In the API documentation, in addition to the note, I would indicate that `otPlatRadioFrameUpdated()` does not read/write any state within OpenThread.  I would also repeat the same comment before the implementation of `otPlatRadioFrameUpdated()` here.

Reply:
OK,  will update it according to your suggestion, thanks.

Reply:
@jwhui The PR is updated, so then ```otPlatRadioFrameUpdated()``` will call ```ProcessTransmitAesCcm()``` function only to perform AES-CCM.